### PR TITLE
Yuidocjs integration

### PR DESCRIPTION
**Pull request by FabianFrank**
Migrated from Python yuidoc to YUIDoc.js, also added a --server switch for the new server feature of YUIDoc.js. Also removed all async shell calls from the doc generation process and replaced them with sync syscalls.

Utilized https://github.com/ryanmcgrath/wrench-js to avoid reinventing the wheel when re-implementing recursive fs operations.

This pull request resolves issue #237 and issue #84.

### CODE REVIEW

File: source/package.json [hunk 1]
```diff
@@ -20,7 +20,9 @@
         "express": "2.5.2",
         "mime": "1.2.4",
         "yui": "~3.5.1",
-        "yuitest": "~0.7.4"
+        "yuitest": "~0.7.4",
+        "wrench": "~1.3.9",
```
Comment (drewfish) [inline source/package.json]: I wonder if "wrench" and "yuidocjs" should be in "devDependencies" instead?  (That is, neither of those are needed in production.)

Hmmm... perhaps there should be a separate "mojito-dev" (or "mojito-tools") package that contains development-time things (and let "mojito" just contain the runtime parts).

Comment (drewfish) [inline source/package.json]: Oh yeah, for sanity please keep the dependencies sorted.

Comment (FabianFrank) [inline source/package.json]: Those are good points. I did not realize that the dependencies were sorted, sorry. I agree with your assessment and I will move them to devDependencies.

**Comment by isao** [comment]
secondary failed (pull job 30) because wrench isn't whitelisted internally

even so :+1: :+1:

**Comment by FabianFrank** [comment]
I requested wrench to be whitelisted internally.

**Comment by FabianFrank** [comment]
@isao wrench has been whitelisted internally, so the secondary build should also pass now. can you please check again? thanks!

**Comment by isao** [comment]
nice. mridgeway's processing PRs this sprint, assigning..

Fabian, can you give him a heads up on any license/manifest/readme changes we need as well?

**Comment by FabianFrank** [comment]
I don't think we have to make changes to the README or other documentation, because the usage of mojito docs didn't change. wrench is MIT licensed, like all other npm modules that we use. We could mention it at the bottom of the README, like we mention Mustache. On the other hand we also don't mention other dependencies there, so I wasn't sure if we want this. Any thoughts?

**Comment by FabianFrank** [comment]
@isao @mridgway Is there an ETA for when we can merge this one?

**Comment by mridgway** [comment]
@FabianFrank I'll get this fixed up and in by the end of the week.

**Comment by mridgway** [comment]
Merged. Thanks a bunch @FabianFrank.

**Pull request merged**